controllers/storagecluster/cephcluster_test.go

```diff
@@ -671,3 +674,103 @@ func TestNewCephDaemonResources(t *testing.T) {
 		assert.Equalf(t, c.expected, got, c.name)
 	}
 }
+
+func TestGetNetworkSpec(t *testing.T) {
+	testTable := []struct {
+		desc     string
+		scSpec   ocsv1.StorageClusterSpec
+		expected cephv1.NetworkSpec
+	}{
+		{
+			desc: "hostNetwork specified as true, network unspecified",
+			scSpec: ocsv1.StorageClusterSpec{
+				HostNetwork: true,
+			},
+			expected: cephv1.NetworkSpec{
+				HostNetwork: true,
+			},
+		},
+		{
+			desc: "hostNetwork specified as false, network unspecified",
+			scSpec: ocsv1.StorageClusterSpec{
+				HostNetwork: false,
+			},
+			expected: cephv1.NetworkSpec{
+				HostNetwork: false,
+			},
+		},
+		{
+			desc: "hostNetwork specified as true, network specified without hostnetwork",
+			scSpec: ocsv1.StorageClusterSpec{
+				HostNetwork: true,
+				Network: &cephv1.NetworkSpec{
+					HostNetwork: false, // same as default
+					IPFamily:    cephv1.IPv6,
+				},
+			},
+			expected: cephv1.NetworkSpec{
+				HostNetwork: true,
+				IPFamily:    cephv1.IPv6,
+			},
+		},
+		{
+			desc: "hostNetwork specified as false, network specified with hostnetwork",
+			scSpec: ocsv1.StorageClusterSpec{
+				HostNetwork: false,
+				Network: &cephv1.NetworkSpec{
+					HostNetwork: true,
+					IPFamily:    cephv1.IPv6,
+					DualStack:   true,
+				},
+			},
+			expected: cephv1.NetworkSpec{
+				HostNetwork: true,
+				IPFamily:    cephv1.IPv6,
+				DualStack:   true,
+			},
+		},
+		{
+			desc: "hostNetwork specified as false, network specified without hostnetwork",
+			scSpec: ocsv1.StorageClusterSpec{
+				HostNetwork: false,
+				Network: &cephv1.NetworkSpec{
+					HostNetwork: false,
+					IPFamily:    cephv1.IPv4,
+				},
+			},
+			expected: cephv1.NetworkSpec{
+				HostNetwork: false,
+				IPFamily:    cephv1.IPv4,
+			},
+		},
+		{
+			desc:     "hostNetwork unspecified, network unspecified",
+			scSpec:   ocsv1.StorageClusterSpec{},
+			expected: cephv1.NetworkSpec{},
+		},
+		{
+			desc: "hostNetwork unspecified, network specified",
+			scSpec: ocsv1.StorageClusterSpec{
+				Network: &cephv1.NetworkSpec{
+					HostNetwork: true,
+					IPFamily:    cephv1.IPv4,
+				},
+			},
+			expected: cephv1.NetworkSpec{
+				HostNetwork: true,
+				IPFamily:    cephv1.IPv4,
+			},
+		},
+	}
+	for _, testCase := range testTable {
+		t.Logf("running testCase %q", testCase.desc)
+		sc := ocsv1.StorageCluster{
+			Spec: testCase.scSpec,
+		}
+		actual := getNetworkSpec(sc)
+		// test Provider, Selectors, HostNetwork, IPFamily, Dualstack
+		diff := cmp.Diff(testCase.expected, actual)
```

Comment:
Why are you explicitly doing a diff instead of `assert.Equal()`? Example: https://github.com/openshift/ocs-operator/blob/master/controllers/storagecluster/cephcluster_test.go#L114

Reply:
...oh, I just learned about `cmp`! https://pkg.go.dev/github.com/google/go-cmp/cmp

That said, you could probably use `assert.True(t, cmp.Equal(a, b), cmp.Diff(a ,b))` or something to that effect, to at least keep things more consistent with how the rest of the test suite currently works.

We should probably look into switching from `reflect` to `cmp` in the code base, as a future PR.

Reply:
Learned about it from looking at rook code! Changing to that format.